tasks/ssl/Linux.yml

```
@@ -0,0 +1,32 @@
+---
+- name: Switch to https
+  set_fact:
+    minio_connection_type: https
+
+- name: Copy key file
+  copy:
+    src: '{{ minio_local_key_path }}'
+    dest: '{{ minio_key_path }}/private.key'
+    owner: '{{ minio_user }}'
+    group: '{{ minio_user }}'
+    mode: 0644
+  become: True
+  when: minio_local_certs
+  notify:
+    - Restart Minio
+
+- name: Copy cert file
+  copy:
+    src: '{{ minio_local_cert_path }}'
+    dest: '{{ minio_cert_path }}/public.crt'
+    owner: '{{ minio_user }}'
+    group: '{{ minio_user }}'
+    mode: 0644
+  become: True
+  when: minio_local_certs
+  notify:
+    - Restart Minio
+
+- name: Generate selfsigned certs
+  include_tasks: selfsigned.yml
+  when: minio_enable_ssl and not minio_local_certs
```

```suggestion
  when: not minio_local_certs
```
minio_enable_ssl is already handled on include_tasks step

---

Accepted